jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/wizard/SchemaSelectionPanel.kt

```diff
@@ -17,23 +17,33 @@ import software.aws.toolkits.jetbrains.services.lambda.sam.SamSchemaDownloadPost
 import software.aws.toolkits.jetbrains.services.schemas.SchemaCodeLangs
 import software.aws.toolkits.jetbrains.ui.connection.AwsConnectionSettingsSelector
 import software.aws.toolkits.resources.message
+import java.util.concurrent.atomic.AtomicBoolean
 import javax.swing.JComponent
 
 /*
  * A panel encapsulating  AWS credential selection during SAM new project creation wizard
   */
 class SchemaSelectionPanel : WizardFragment {
-    private val schemaSelector = SchemaResourceSelector()
-    private val awsConnectionSelector = AwsConnectionSettingsSelector(
-        null,
-        schemaSelector::reloadSchemas
-    )
-    private val component = panel {
-        row {
-            awsConnectionSelector.selectorPanel()(grow)
+    /*
+     * We only want to initialize the selector once it is applicable, so we have to keep track of it
+     */
+    private val selectorInitialized = AtomicBoolean(false)
+    private val schemaSelector by lazy { SchemaResourceSelector() }
```

Comment:
can we just load it here?

Reply:
load what here? the selector? We don't want it to load if the template does not support schemas?

Reply:
I forgot my own logic -.-

The component method isnt gated by isApplicable as I thought it was.

Reply:
Can we do the following?

Change SamInitSelectionPanel.kt:

```
        wizardFragments.forEach { (wizardFragment, jComponent) ->
            wizardFragment.updateUi(projectLocation, selectedRuntime?.runtimeGroup, selectedTemplate)
            jComponent.isVisible = wizardFragment.isApplicable(selectedTemplate)
        }
```
to

```
        wizardFragments.forEach { (wizardFragment, jComponent) ->
            val isApplicable = wizardFragment.isApplicable(selectedTemplate)
            if (isApplicable) {
                wizardFragment.updateUi(projectLocation, selectedRuntime?.runtimeGroup, selectedTemplate)
            }
            jComponent.isVisible = isApplicable
        }
```

and then do the `reload` call in the updateUi method?

Reply:
Also making the vals lazy does not seem to do much given we eagerly load each WizardFragment into the UI and then hide it after

Reply:
Making the vals lazy means they aren't initialized when the wizard is opened, otherwise even if you want to make a java project (as an example) they still get loaded

Reply:
the change in the wizard itself simplifies the logic

Reply:
Ah, didn't realize the wizard init'ed our stuff even if it wasn't selected.